src/cocaine-app/__init__.py

```diff
@@ -214,47 +212,48 @@ def init_minions():
     return m
 
 
-def init_planner(job_processor, niu, namespaces_settings, move_planner, external_storage_converting_planner, lrc_reserve_group_planner):
+def init_planner(job_processor, niu, namespaces_settings):
+
+    planner_new = config.get('scheduler', {}).get('enabled', False)
+
     planner = Planner(meta_db, niu, job_processor, namespaces_settings)
     register_handle(planner.restore_group)
     register_handle(planner.move_group)
     register_handle(planner.move_groups_from_host)
     register_handle(planner.restore_groups_from_path)
     register_handle(planner.ttl_cleanup)
 
-    if move_planner:
+    if planner_new:
+        # redefine planner, while handlers are still stored within old planner object
+        planner = Scheduler(meta_db, niu, job_processor, namespaces_settings)
```

Comment:
Done in https://github.com/yandex/mastermind/pull/417/commits/879ce74554565fdfc93ed5cf8a914b829944d6d8